### PR TITLE
[CSL-1550] Add --behavior

### PR DESCRIPTION
**Pull request by neongreen**
Note that the first part of CSL-1550 was (improperly) merged in a different branch: https://github.com/input-output-hk/cardano-sl/pull/1130/commits/ea0f06c6009896ebfc75c4ee68e9f122bd67b1e8.

### CODE REVIEW

File: core/Pos/Aeson/Crypto.hs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import           Crypto.Hash   (HashAlgorithm)
+import           Data.Aeson    (FromJSON (..), ToJSON (..))
+import           Formatting    (sformat)
+
+import           Pos.Crypto    (AbstractHash, PublicKey, fullPublicKeyF, hashHexF)
+import           Pos.Util.Util (parseJSONWithRead)
+
+instance ToJSON (AbstractHash algo a) where
+    toJSON = toJSON . sformat hashHexF
```
Comment (georgeee) [inline core/Pos/Aeson/Crypto.hs]: why not `show` here (for implementations to be consistent)?

Comment (neongreen) [reply core/Pos/Aeson/Crypto.hs]: This isn't my code, I just moved it.

File: core/cardano-sl-core.cabal [hunk 2]
```diff
@@ -150,32 +143,42 @@ library
...
                      , pvss
                      , quickcheck-instances
                      , random
+                     , reflection
+                     , resourcet
                      , safecopy
                      , semigroups
                      , serokell-util
                      , stm
+                     , stm
```
Comment (domenkozar) [inline core/cardano-sl-core.cabal]: double?

Comment (neongreen) [reply core/cardano-sl-core.cabal]: Bad merge, will fix it.

**Pull request merged**